src/SignalRServiceExtension/Config/SignalRFunctionsHostBuilderExtensions.cs

```diff
@@ -0,0 +1,95 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Azure.Functions.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.IdentityModel.Tokens;
+
+namespace Microsoft.Azure.WebJobs.Extensions.SignalRService
+{
+    using SignalRConnectionInfoConfigureFunc = Func<AccessTokenResult, HttpRequest, SignalRConnectionDetail, SignalRConnectionDetail>;
+
+    // todo: add more DI
+    public static class SignalRFunctionsHostBuilderExtensions
+    {
+        public static IFunctionsHostBuilder AddAuth(this IFunctionsHostBuilder builder, Action<TokenValidationParameters> configureTokenValidationParameters)
```

Comment:
Wonder if this should be `AddSignalRAuth` (or `AddSignalRAuthentication`) to make it clear in when used that it's for SignalR.